packages/http/src/validator/utils/wildcardMediaTypeMatch.ts

```diff
@@ -0,0 +1,45 @@
+import { parse, ParsedMediaType } from 'content-type';
+import * as O from 'fp-ts/Option';
+import { pipe } from 'fp-ts/function';
+
+export function wildcardMediaTypeMatch(mediaTypeA: string, mediaTypeB: string) {
+  return pipe(
+    O.Do,
+    O.bind('a', () => parseContentType(mediaTypeA)),
+    O.bind('b', () => parseContentType(mediaTypeB)),
+    O.fold(
+      () => false,
+      ({ a, b }) => {
+        return a.type === b.type && a.subtype === b.subtype;
+      }
+    )
+  );
+}
+
+const REGEXP = /^([-\w.]+)\/(?:([-\w.]+)\+)?([-\w.]+)$/;
```

Comment:
1. I think there are some characters allowed by the standard that are not included by `[-\w.]`, such as `!`, `#` and a few others.  See [RFC-6838 §4.2](https://datatracker.ietf.org/doc/html/rfc6838#section-4.2).
1. Is this intended to be more liberal than the media-type registries for the top-level type?  That is, are completely custom top-level types allowed?  If not, I think the list of top-level types could be included exhaustively (based on "Registries included below" [here](https://www.iana.org/assignments/media-types/media-types.xhtml).
2. I think it should be possible to include the "parameter" portion of a media type following a semicolon (e.g., `charset=UTF-8`).  Is that possible here?  I wasn't able to figure out if that had been removed prior to this code running.

Reply:
This is a good point! @EdVinyard I found a regex at Stackoverflow that based on the intersection of the grammar from [RFC 2045 Section 5.1](https://www.rfc-editor.org/rfc/rfc2045#section-5.1) and [RFC 7231 Section 3.1.1.1](https://www.rfc-editor.org/rfc/rfc7231#section-3.1.1.1). They both related to the [RFC-6838](https://datatracker.ietf.org/doc/html/rfc6838#section-4.2) as far as I can tell.

This regex should be able to captures type, subtype, and parameters: 

```js
/(application|audio|font|example|image|message|model|multipart|text|video|x-(?:[0-9A-Za-z!#$%&'*+.^_`|~-]+))/([0-9A-Za-z!#$%&'*+.^_`|~-]+)((?:[ \t]*;[ \t]*[0-9A-Za-z!#$%&'*+.^_`|~-]+=(?:[0-9A-Za-z!#$%&'*+.^_`|~-]+|\"(?:[^\"\\\\]|\\.)*\"))*)/
```

Looks a bit scary to be honest haha.

Reply:
Thanks for regexp. However I don't need this since first part of parsing content-type over here is done by `content-type` package that already has this regexp, therefore I don't need to repeat it. What I need to do is to extract suffix and subtype. 

However @EdVinyard is right that I didn't cover enough characters here. Updated Regexp